### PR TITLE
8264102: JTable Keyboards Navigation differs with Test Instructions.

### PR DESCRIPTION
**Pull request by TejeshR13**
Updated the actions for "ctrl shift DOWN" to "`selectLastRowExtendSelection`" and  "ctrl shift UP", "`selectFirstRowExtendSelection`" in Basic and other Look and feel. Tested in CI and no regressions found. 
Test to be used - [javax/swing/JTable/KeyBoardNavigation/KeyBoardNavigation.java](https://github.com/openjdk/jdk/blob/master/test/jdk/javax/swing/JTable/KeyBoardNavigation/KeyBoardNavigation.java)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8264102](https://bugs.openjdk.org/browse/JDK-8264102): JTable Keyboards Navigation differs with Test Instructions. (**Bug** - P3)

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) ⚠️ Review applies to [ccd5fd5b](https://git.openjdk.org/jdk/pull/17236/files/ccd5fd5b61601dbf7c5559aa7d45ff8056e131cd)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17236/head:pull/17236` \
`$ git checkout pull/17236`

Update a local copy of the PR: \
`$ git checkout pull/17236` \
`$ git pull https://git.openjdk.org/jdk.git pull/17236/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17236`

View PR using the GUI difftool: \
`$ git pr show -t 17236`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17236.diff">https://git.openjdk.org/jdk/pull/17236.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17236#issuecomment-1874832866)

### CODE REVIEW

**Comment by prsadhuk** [comment]
It is definitely required for Windows L&F as excel also has same functionality for CTRL+SHIFT+UP/DOWN but I amnot sure if spec mandates it for all L&F...Did you see anywhere in spec that ctrl shift up/down should select all populated cells in the column.
Please check in linux and macos if native spreadsheet has same functionality, if not restrict it to Windows only...

Also, add bugid to the existing jtreg test...it may be recommended to modify the deprecated applet based test, since you are relying on that test to test this fix, to normal manual PassFailJFrame based test

**Comment by TejeshR13** [comment]
> It is definitely required for Windows L&F as excel also has same functionality for CTRL+SHIFT+UP/DOWN but I amnot sure if spec mandates it for all L&F...Did you see anywhere in spec that ctrl shift up/down should select all populated cells in the column. Please check in linux and macos if native spreadsheet has same functionality, if not restrict it to Windows only...
> 
> Also, add bugid to the existing jtreg test...it may be recommended to modify the deprecated applet based test, since you are relying on that test to test this fix, to normal manual PassFailJFrame based test

I couldn't trace out to the spec and same functionality is seen in linux and macos too.. Sure update the test too.

File: test/jdk/javax/swing/JTable/KeyBoardNavigation.java [hunk 1]
```diff
@@ -146,25 +139,66 @@ public void setValue(Object value) {
...
+
+        JTable tableView = new JTable(dataModel);
+        // Turn off auto-resizing so that we can set column sizes programmatically.
+        // In this mode, all columns will get their preferred widths, as set blow.
+        tableView.setAutoResizeMode(JTable.AUTO_RESIZE_OFF);
+        return tableView;
+    }
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JTable/KeyBoardNavigation.java:194]: The fix is for all L&F but the test is testing only Metal...It should be given chance to use default L&F in each platform..
If 4112270 is only about Metal, then a new test should be spawned to test this fix for all fixed L&F..

Comment (TejeshR13) [reply test/jdk/javax/swing/JTable/KeyBoardNavigation.java:194]: 4112270 is not only about Metal L&F. In Aqua L&F the functionality is different, it enables the cell to edit mode. Yet, in macos excel functionality is same as that of windows. Should we update Aqua too?

Comment (prsadhuk) [reply test/jdk/javax/swing/JTable/KeyBoardNavigation.java:194]: Which macos excel application you tested? Microsoft excel or NUmbers...I guess Numbers should be ideal native application to tell if native OS does need it and depending on that, we need to change it accordingly for Aqua L&F..

Comment (TejeshR13) [reply test/jdk/javax/swing/JTable/KeyBoardNavigation.java:194]: In both Microsoft excel and Numbers, Ctrl shift UP/DOWN extends till first/end row.

**Review by prsadhuk — APPROVED** [review]
Looks ok to me...
Only thing is...If it could be done without much trouble, I would have listed the combinations of keyboard navigation in the test itself instead of directing to 4112270

**Comment by TejeshR13** [comment]
> Looks ok to me... Only thing is...If it could be done without much trouble, I would have listed the combinations of keyboard navigation in the test itself instead of directing to 4112270

Sure, can do that.

File: test/jdk/javax/swing/JTable/KeyBoardNavigation.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+            public void setValueAt(Object aValue, int row, int column) {
+                System.out.println("Setting value to: " + aValue);
+                data[row][column] = aValue;
+            }
+        };
+
+        JTable tableView = new JTable(dataModel);
+        // Turn off auto-resizing so that we can set column sizes programmatically.
+        // In this mode, all columns will get their preferred widths, as set blow.
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTable/KeyBoardNavigation.java:221]: looks like a typo
```suggestion
        // In this mode, all columns will get their preferred widths, as set below.
```

File: test/jdk/javax/swing/JTable/KeyBoardNavigation.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        tableView.setAutoResizeMode(JTable.AUTO_RESIZE_OFF);
+        return tableView;
+    }
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            try {
+                initTest();
+            } catch (Exception e) {
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTable/KeyBoardNavigation.java:230]: try-catch block may be removed.

Comment (TejeshR13) [reply test/jdk/javax/swing/JTable/KeyBoardNavigation.java:230]: It is required since I'm using PassFailJFrame inside `initTest()`.

**Review by kumarabhi006 — APPROVED** [review]
LGTM.

**Pull request closed without merging**